pallets/pool-system/src/lib.rs

```diff
@@ -599,13 +624,28 @@ pub mod pallet {
 					Error::<T>::MinEpochTimeHasNotPassed
 				);
 
-				let (nav, nav_last_updated) = T::NAV::nav(pool_id).ok_or(Error::<T>::NoNAV)?;
+				// Get positive NAV from AUM
+				let (nav_aum, aum_last_updated) =
+					T::AssetsUnderManagementNAV::nav(pool_id).ok_or(Error::<T>::NoNAV)?;
+				ensure!(
+					now.saturating_sub(aum_last_updated) <= pool.parameters.max_nav_age,
+					Error::<T>::NAVTooOld
+				);
 
+				// Calculate fees to get negative NAV
+				T::OnEpochTransition::on_closing_mutate_reserve(
+					pool_id,
+					nav_aum,
+					&mut pool.reserve.total,
+				)?;
+				let (nav_fees, fees_last_updated) =
+					T::PoolFeesNAV::nav(pool_id).ok_or(Error::<T>::NoNAV)?;
 				ensure!(
-					now.saturating_sub(nav_last_updated) <= pool.parameters.max_nav_age,
+					now.saturating_sub(fees_last_updated) <= pool.parameters.max_nav_age,
 					Error::<T>::NAVTooOld
 				);
 
+				let nav = nav_aum.saturating_sub(nav_fees);
```

Comment:
CR:

The following three lines need to change. Otherwise, the NAV is not correct. 

from:
```rust 
let nav = nav_aum.saturating_sub(nav_fees);
let submission_period_epoch = pool.epoch.current;
let total_assets = nav.ensure_add(pool.reserve.total)?;
```

to:
```rust 
let positive_nav = nav_aum.ensure_add(pool.reserve.total)?;
let nav = positive_nav.ensure_sub(nav_fees).ok_or(Error::<T>::NegativeBalanceSheet);
let submission_period_epoch = pool.epoch.current;
```

In the above `total_assets` is `nav`, which is the actual correct term, but we are not using it correctly in the codebase. Feel free to leave it with `total_assets`.